src/main/java/com/vmware/vipclient/i18n/base/cache/PatternCacheItem.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2019-2020 VMware, Inc.
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package com.vmware.vipclient.i18n.base.cache;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class PatternCacheItem extends FormatCacheItem implements CacheItem {
+
+    private final Map<String, Object> cachedData = new HashMap<String, Object>();
+
+    public PatternCacheItem() {
+
+	}
+
+	public PatternCacheItem(Map<String, Object> dataMap, String etag, long timestamp, Long maxAgeMillis) {
+		this.set(dataMap, etag, timestamp, maxAgeMillis);
+	}
+
+    public synchronized void set(Map<String, Object> dataMap, long timestamp) {
+        this.set(dataMap, null, timestamp, null);
+    }
+
+    public synchronized void set(Map<String, Object> dataToCache, String etag, long timestamp, Long maxAgeMillis) {
+        if (dataToCache != null)
+            this.cachedData.putAll(dataToCache);
+        this.set(etag, timestamp, maxAgeMillis);
+    }
+
+    public synchronized void set(String etag, long timestamp, Long maxAgeMillis) {
+        if (etag != null && !etag.isEmpty())
+            setEtag(etag);
+        setTimestamp(timestamp);
+        if (maxAgeMillis != null)
+            setMaxAgeMillis(maxAgeMillis);
+    }
+
+    public synchronized void set(PatternCacheItem cacheItem) {
+        this.set(cacheItem.getCachedData(), cacheItem.getEtag(), cacheItem.getTimestamp(), cacheItem.getMaxAgeMillis());
+    }
```

Comment:
I remove this method as it's not used.
But I wonder the CacheItem properties can be modified individually even if CacheItem is retrieved from cache? Because no individually setter methods are provided.

Reply:
Same reply as I have in LocaleCacheItem: _"Hmm, I think I do not understand your comment. The point is to NOT allow individual modification of these properties, so 1 thread will lock the object first, set all three, then release the lock. These props should always be updated as a set."_